cf-java-monitoring-custom-metrics-clients/cf-custom-metrics-clients-java/src/main/java/com/sap/cloud/cf/monitoring/java/converter/MetricConverter.java

```diff
@@ -0,0 +1,43 @@
+package com.sap.cloud.cf.monitoring.java.converter;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import com.sap.cloud.cf.monitoring.client.model.Metric;
+
+public abstract class MetricConverter<T> {
+    private static final String KEY_TYPE = "type";
+
+    public enum MetricType {

```

Comment:
This is a strange design choice, to have an explicit enum of all the derived classes in the base class. So basically this class knows about all its derivatives. Adding a new MetricType not only requires a new class, but also a change to this enum.

Reply:
Yes, that was the idea.